extension/headerssetter/source/context.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/headerssetter/source"
```

Comment:
Perhaps this should be in an internal package? Like `github.com/open-telemetry/opentelemetry-collector-contrib/extension/headerssetter/internal/source`. Otherwise, Source, ContextSource and others would be public to the world.